app/src/org/commcare/models/database/app/AppDatabaseUpgrader.java

```diff
@@ -1,20 +1,31 @@
 package org.commcare.models.database.app;
 
 import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
 import android.util.Log;
 
 import net.sqlcipher.database.SQLiteDatabase;
 
+import org.apache.commons.lang3.StringUtils;
+import org.commcare.android.database.app.models.FormDefRecord;
+import org.commcare.android.database.app.models.InstanceRecord;
+import org.commcare.android.database.app.models.UserKeyRecord;
+import org.commcare.android.database.app.models.UserKeyRecordV1;
+import org.commcare.android.resource.installers.XFormAndroidInstaller;
+import org.commcare.android.resource.installers.XFormAndroidInstallerV1;
+import org.commcare.android.storage.framework.Persisted;
 import org.commcare.engine.resource.AndroidResourceManager;
-import org.commcare.modern.database.TableBuilder;
+import org.commcare.models.AndroidPrototypeFactoryV1;
 import org.commcare.models.database.ConcreteAndroidDbHelper;
 import org.commcare.models.database.DbUtil;
 import org.commcare.models.database.SqlStorage;
-import org.commcare.android.database.app.models.UserKeyRecord;
-import org.commcare.android.database.app.models.UserKeyRecordV1;
 import org.commcare.models.database.migration.FixtureSerializationMigration;
-import org.commcare.android.storage.framework.Persisted;
+import org.commcare.modern.database.TableBuilder;
+import org.commcare.provider.FormsProviderAPI;
+import org.commcare.provider.InstanceProviderAPI;
 import org.commcare.resources.model.Resource;
+import org.javarosa.core.util.externalizable.PrototypeFactory;
```

Comment:
I think that this file needs to have all of the upgradeXtoY methods from 1 to 8 audited to make sure their records use the old version of the classes. We need to load old Resources with the V1 installer for XForms, for example, right?

Reply:
Good point though none of these methods seem to be loading Resources i.e. performing a read on Resource table!